src/games/strategy/triplea/ai/proAI/util/ProTransportUtils.java

```diff
@@ -162,6 +162,25 @@ public int findUnitsTransportCost(final List<Unit> units) {
     return transportCost;
   }
 
+  public List<Unit> getAirThatCantLandOnCarrier(final PlayerID player, final Territory t, final List<Unit> units) {
+    final GameData data = ai.getGameData();
+    int capacity = AirMovementValidator.carrierCapacity(units, t);
```

Comment:
Okay, then probably placing it a line lower would have been slightly better style. The final variables would get the longer scope, and a reader would be more clear that this wasn't an inconsistency between marking variable final. 

Having the variable closer to the loop also helps a reader guess that we'll be modifying in in the loop. Letting your readers make obvious guesses and be correct is how you get very understandable code. 

Here though the variable name lends to the misunderstanding.  "capacity" for a carrier is a fixed value, and the variable name "capacity" sounds like a fixed value. So the guess is then it would be a fixed value and then the reader is left why it is not also final (so either it was a mistake, or now the reader needs to scan this method specifically looking at this variable in specific  to understand what *this* line is really doing).

So moving the variable down a line, call it "capacityLeft", and a reader will always understand what is happening on the first pass, no need scan this method multiple times to understand it.